### PR TITLE
DHFPROD-7204: Popover Search Functionality

### PR DESCRIPTION
**Pull request by rahulvudutala**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java [hunk 1]
```diff
@@ -33,6 +34,7 @@
     JobController jobController;
 
     private final static String STEP_RESPONSES_PATH = "/api/jobs/stepResponses";
+    private final static String GET_MATCHING_VALUES_PATH = "/api/jobs/stepResponses/facet-values";
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java:37]: Seeing "stepResponses" and "facet-values" begs the question - why is one camelCase and the other is worm-case? I'd think "facetValues" is better for consistency.

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java [hunk 2]
```diff
@@ -85,4 +87,28 @@ public void testFindStepResponses() throws Exception {
...
         verifyRequestIsForbidden(buildJsonPost(STEP_RESPONSES_PATH, json));
     }
+
+    @Test
+    public void testGetMatchingPropertyValues() throws Exception {
+        String json = "{\n" +
+                "  \"facetName\": \"flowName\",\n" +
+                "  \"searchTerm\": \"simple\"\n" +
+                "}";
+        runAsAdmin();
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java:97]: You can install a project as a developer, so it's better to use runAsDataHubDeveloper

File: marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java [hunk 3]
```diff
@@ -85,4 +87,28 @@ public void testFindStepResponses() throws Exception {
...
+        installReferenceModelProject(true);
+
+        loginAsTestUserWithRoles("hub-central-operator");
+        runFlow(new FlowInputs("simpleMapping"));
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/JobControllerTest.java:101]: You can use runSuccessfulFlow which will verify that the flow ran successfully

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/job-query-lib.sjs [hunk 4]
```diff
@@ -141,6 +141,26 @@ function buildFacetQueries(whereClause) {
   return queries;
 }
 
+function getMatchingPropertyValues(facetValuesSearchQuery) {
+  const facetName = facetValuesSearchQuery.facetName;
+  const searchTerm = facetValuesSearchQuery.searchTerm;
+  const limit = facetValuesSearchQuery.limit ? facetValuesSearchQuery.limit : 10 ;
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/job-query-lib.sjs:147]: I don't see a test for limit; you could do this pretty easily by having 2 different values, and verify you get both back without a limit, and then submit a limit of 1 to verify you get 1 back.

**Review by rjrudin — CHANGES_REQUESTED** [review]

**Review by rjrudin — APPROVED** [review]

**Review by Sanjeevani19 — DISMISSED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Review by ryanjdew — APPROVED** [review]

**Pull request merged**